src/proto/mod.rs

```diff
@@ -40,3 +41,50 @@ impl<'a, T: Serialize + Deserialize<'a>> Blob for T {
         from_bytes(blob)
     }
 }
+
+pub mod recursive {
+    use super::{from_bytes, to_bytes};
+    use serde::{
+        de::{self, Deserializer, Visitor},
+        ser::Serializer,
+        Deserialize, Serialize,
+    };
+    use std::{fmt, marker::PhantomData};
+
+    pub fn serialize<T, S>(obj: &T, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        T: Serialize,
+        S: Serializer,
+    {
+        serializer.serialize_bytes(&to_bytes(obj).unwrap())
```

Comment:
:thinking: The `unwrap` is making me dizzy. Maybe let's map the error or if it's really not going to happen at least make it an `.expect` call with explicit message...?